assets/airports/essa.json

```diff
@@ -0,0 +1,3558 @@
+            "holds": {
+                "XILAN": "069|left|1.5min|S230-",
+                "TEB"  : "178|right|1min|S230-"
+            },
+            "entryPoints": {
+                "XILAN": [["XILAN", "A150-"]]
+            },
+            "body": ["SA570", "@TEB"],
+            "rwy": {
+                "ESSA08": []
+            },
+            "draw": [
+                ["XILAN", "SA570", "TEB"]
+            ]
+        },
+        "XILAN5P": {
+            "icao": "XILAN5P",
+            "name": "Xilan Five Papa",
+            "holds": {
+                "XILAN": "069|left|1.5min|S230-",
+                "ERK"  : "038|left|1min|S230-"
+            },
+            "entryPoints": {
+                "XILAN": [["XILAN", "A150-"]]
+            },
+            "body": ["SA570", "TEB", "@ERK"],
+            "rwy": {
+                "ESSA19R": [],
+                "ESSA19L": []
+            },
+            "draw": [
+                ["XILAN", "SA570", "TEB", "ERK"]
+            ]
+        },
+        "XILAN5T": {
+            "icao": "XILAN5T",
+            "name": "Xilan Five Tango",
+            "holds": {
+                "XILAN": "069|left|1.5min|S230-",
+                "TEB"  : "178|right|1min|S230-"
+            },
+            "entryPoints": {
+                "XILAN": [["XILAN", "A120-"]]
+            },
+            "body": ["SA570", "@TEB"],
+            "rwy": {
+                "ESSA26": []
+            },
+            "draw": [
+                ["XILAN", "SA570", "TEB"]
+            ]
+        }
+    },
+    "_comment": "Data for spawnpatterns is 2018/2019 (pre-covid) statistics from https://en.wikipedia.org/wiki/Stockholm_Arlanda_Airport",
+    "_comment": "London example:",
+    "_comment": "a. ['Passengers handled 2019' 1420655] / ['Total passenger traffic 2017' 26642034] = ['Passenger proportion to/from London' 0.0533] (using 2017 total because 2019 is not available, should be quite representative)",
+    "_comment": "b. ['Total landings 2017' 122697] * ['Passenger proportion to/from London' 0.0533] = ['Landings from London' 6540] (using 2017 total because 2019 is not available)",
+    "_comment": "c. ['Landings from London' 6540] / ['Hours in a year' 8760] / (2/3) = ['Aircraft per hour from London' 1.1] (2/3 because I assume almost no passenger traffic during the night 1/3 of each day)",
+    "_comment": "d. Divide the 1.1 rate between London/Heathrow and London/Gatwick by feeling. This gives me 0.7 to Heathrow and 0.4 to Gatwick",
+    "_comment": "e. Find which airlines operate these destinations and guess the weights by feeling",
+    "_comment": "f. Find a reasonable route from London to Arlanda on http://flightplandatabase.com and select a reasonable STAR",
+    "_comment": "g. Create a spawnPatterns for arrivals from Heathrow and Gatwick using the above information",
+    "_comment": "h. Find a reasonable SID and create corresponding spawnPatterns for departures (assuming landing rate and takeoff rate are equal)",
```

Comment:
Thoughts on this procedure? Does it seem somewhat accurate?

Reply:
To be honest, there is not one way of getting spawnPattern data. Everyone has a different method. Mine changes depending on the airport. What you describe there sounds perfectly fine. Other options you can look into:

Sometimes (though rarely) the national aviation agency will publish a report which includes statistics about a year's most flown routes sorted by airlines or aircraft.

Wikipedia articles sometimes contain frequency tables for airline destinations. Often these can only be found in one specific language, so you can try switching languages.

There's a tool written by Indianbhaji that takes data from the last 24hrs of Flight Radar and gives you the frequencies of different airlines and aircraft.

You can manually sort through pages of flightaware and make a tally of airlines.

All of these methods have their imperfections, particularly given the decreased traffic volumes due to a certain global emergency. In terms of the SIDs and STARs, the method you describe is pretty much the only one available.

Reply:
The data on Wikipedia is the same data published by Swedavia, just formatted a bit better. I could only find statistics based on passenger number, not departures. The only thing relating to aircraft was total number of landings, so I guess this method will have to do.

I was thinking about looking at the arrival/departure data but I couldn't find any history anywhere, to avoid the Covid-decrease on traffic, so I ruled out that idea.

Reply:
I'll resolve this conversation because I'm satisfied with using this method. Furthermore, I will combine traffic from different airports in approximately the same direction into fewer spawnPatterns as discussed in issue #1762